src/Interpreters/Access/InterpreterGrantQuery.cpp

```diff
@@ -190,7 +190,7 @@ namespace
         /// REVOKE SELECT ON system.* FROM user2;
         ///
         /// the query `REVOKE SELECT ON *.* FROM user1` executed by user2 should succeed.
-        if (current_user_access.getAccessRights()->containsWithGrantOption(access_to_revoke))
+        if (current_user_access.getAccessRightsWithImplicit()->containsWithGrantOption(access_to_revoke))
```

Comment:
Could you kindly explain the difference between `access` and `access_with_implicit`?

Reply:
We have some implicit grants in our RBAC. For example, when a user has `SELECT` on a table, they should also be able to execute `SHOW CREATE TABLE`. Thus, we add a `SHOW` grant implicitly under the hood.

```

SHOW GRANTS FOR test

Query id: e8d80ef0-f85f-4e75-bd78-e9eeb44ae26c

   ┌─GRANTS FOR test───────────────────────────────┐
1. │ GRANT SELECT ON *.* TO test WITH GRANT OPTION │
   └───────────────────────────────────────────────┘

1 row in set. Elapsed: 0.002 sec. 

pufit-MBP.attlocal.net :) SHOW GRANTS FOR test WITH IMPLICIT

SHOW GRANTS FOR test WITH IMPLICIT

Query id: b1fb45ac-823e-467e-aea3-f64dc0c58060

    ┌─GRANTS FOR test WITH IMPLICIT────────────────────────────────────────────────────────────┐
 1. │ GRANT SHOW DATABASES, SHOW TABLES, SHOW COLUMNS, SELECT ON *.* TO test WITH GRANT OPTION │
    └──────────────────────────────────────────────────────────────────────────────────────────┘

19 rows in set. Elapsed: 0.002 sec. 

pufit-MBP.attlocal.net :)

```

Reply:
Got it. Thanks